### ISSUE TITLE
Poor performance

### ISSUE THREAD

**Issue by tanpro260196** [openappsec/open-appsec-npm#5]
Load time for almost any sites increased by tenfold compared to the original NPM. Regardless of detect/prevent mode and level.

Sometime a site just timeout while initiating connection, especially if the site have a lot of images.

**Comment by orianelou** [comment]
Hi @tanpro260196, 

We're looking into this, could you please share your system's specs (CPU, RAM)? you can also send us the details to info@openappsec.io.

Thank you! and hope to have some good news soon,

Oriane
open-appsec team

**Comment by tanpro260196** [comment]
Hi,

I am running it on an old PC.
CPU: i7 8700 (non-k)
16GB of ram DDR4.

**Comment by tanpro260196** [comment]
Hi there, is there any update?

**Comment by mgrimace** [comment]
Ok adding to this, appsec-agent is consistently running between 200-400% of my cpu according to `docker stats` 
Proxmox, 10 cores, 32 gb ram, Debian 12 LXC, i7 8700

**Comment by mgrimace** [comment]
Follow-up: removing open-appsec and going back to stock NPM dropped my overall Proxmox server CPU use from 40ish% to 8%.

**Comment by jiriteach** [comment]
Yup - I have experienced this as well. Similar processor and have the agent on Docker running ~ 45-50% CPU even when its not doing much. Managed to setup everything as required and works well but the increased CPU is an issue.

**Comment by jiriteach** [comment]
So I've tried the Linux Embedded Agent for comparison and the CPU usage is only ~ 5-8%. Any reason why its so hard on the Docker container version? Anything that can be done to optimise? Might have to use the Linux Embedded Agent and normal NGINX as opposed to NPM.

**Comment by brandonkal** [comment]
Any update on this? Would it be possible to use NPM for nginx and appsec config generation but run nginx on the host?
I am experimenting with using Nomad to run the WAF tasks

**Comment by AnnoyingTechnology** [comment]
Any update on this ? @orianelou

**Comment by orianelou** [comment]
Thanks for reporting this again. Our developers have conducted another investigation and identified an issue that may cause higher-than-usual performance impact for some users. We’ve prioritized the fix, and it’s planned for inclusion in our next release, expected within the next week.

**Comment by luk-ada** [comment]
Waiting for new release. Current setup increases CPU usage by 6-8% on my Intel N100.

**Comment by orianelou** [comment]
Hi @luk-ada, 

An improved version was released last month (version 1.1.21), do you mean the current set up takes 6-8 more the open-appsec on a previous version? or generally 6-8% CPU?

**Comment by luk-ada** [comment]
Hi @orianelou 
It's generally using up to 10%, I'm trying to protect two sites used by two users and it's a bit of load for such small volume. I'm not able to compare with previous versions as I was not using them.

![Image](https://github.com/user-attachments/assets/cca1f9cd-8d4c-4049-a1fe-af71ad71c33f)

**Comment by NolandTech** [comment]
Just uninstalled due to poor cpu preformance. using constant 15-20% then jumping upto 200%. Using with NPMPlus, crowdsec, and modsec. Crowdsec is using max 10% everyso often then going back to idle...

**Comment by orianelou** [comment]
Hi @luk-ada, 

That seems to be withing the expected range.

**Comment by AnnoyingTechnology** [comment]
> Hi [@luk-ada](https://github.com/luk-ada),
> 
> That seems to be withing the expected range.

Could you clarify : is openappsec _expected_ to consumme cpu cycles when **not** analyzing any traffic ?

**Comment by orianelou** [comment]
Hi @AnnoyingTechnology, 

Yes, open-appsec may consume some CPU cycles even when not actively analyzing traffic. This is due to background processes. Also please note that @luk-ada didn't mention that there's no traffic.

**Comment by NolandTech** [comment]
> Hi [@NolandTech](https://github.com/NolandTech),
> 
> Thanks for your feedback! Running multiple WAFs (modsec and open-appsec) on the same machine can lead to resource contention, especially under high traffic loads. Each WAF competes for CPU and memory, which may explain the spikes you're seeing.
> 
> Could you try testing without modsec and see if the CPU usage improves? That would help narrow down the cause. Let us know what you find!

I've disabled Modsec, and have seen little to no improvement.. My services, are also receiving little to no traffic. 

![Image](https://github.com/user-attachments/assets/df197c6c-b5f3-490d-82e0-1e3b548c2cf6)

Disabling all my servies seemd to stop the cpu spikes, but load is pretty heavy still for not doing anything...

![Image](https://github.com/user-attachments/assets/baed9646-b8bb-4bdc-b86e-14ffc8b06c50)

**Comment by AnnoyingTechnology** [comment]
If this has been _completed_ (fixed) please specify in which relevant commit. 

Otherwise, please re-open.

**Comment by Korenchkin** [comment]
i'm too into this strange behavior on npmplus,no data(not set yet,still using caddy),yet ~150% of 2 cores i5-7600k,log file spams this:
```
tail: cannot open '/var/log/nano_agent/cp-nano-http-transaction-handler.log*' for reading: No such file or directory
tail: no files remaining
```
could be relevant :)
